service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterMapper.java

```diff
@@ -53,10 +57,36 @@ public String mapEncounterToEhrComposition(Encounter encounter) {
             .displayName(buildDisplayName(encounter))
             .originalText(buildOriginalText(encounter));
 
+        final Optional<String> recReference = findParticipantWithCoding(encounter, "REC")
+            .map(idMapper::get);
+
+        recReference.ifPresent(encounterStatementTemplateParameters::author);
+
+        final Optional<String> pprfReference = findParticipantWithCoding(encounter, "PPRF")
+            .map(idMapper::getOrNew);
```

Comment:
should be ::get to fail if the reference doesn't exist

Reply:
Did I understand the spreadsheet wrong?
> Populate the participant from the participant of type PPRF and if there is no PPRF participant make the REC participant the PPRF, All other additional participants are ignored